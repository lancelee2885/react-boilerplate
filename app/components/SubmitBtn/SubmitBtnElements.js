import styled from 'styled-components';

export const SubmitBtnWrapper = styled.button`
  margin-top: 32px;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
