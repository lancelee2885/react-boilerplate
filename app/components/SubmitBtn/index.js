import React from 'react';
import { SubmitBtnWrapper } from './SubmitBtnElements';

/** SubmitBtn:
 *  - Simple button for sumbitting a form.
 */
function SubmitBtn() {
  return <SubmitBtnWrapper> Submit </SubmitBtnWrapper>;
}

export default SubmitBtn;
