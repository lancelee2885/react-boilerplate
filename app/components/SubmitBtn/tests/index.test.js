import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import SubmitBtn from '../index';
import configureStore from '../../../configureStore';

describe('<SubmitBtn />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <SubmitBtn />
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
