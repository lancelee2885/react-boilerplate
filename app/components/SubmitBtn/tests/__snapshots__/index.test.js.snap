// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitBtn /> should render and match the snapshot 1`] = `
<button
  className="SubmitBtnElements__SubmitBtnWrapper-sc-15ad74p-0 dSStNU"
>
   
   
</button>
`;
